src/bootstrap.coffee

```diff
@@ -101,6 +148,13 @@ bootstrapper.done = new Promise (resolve) ->
 	bootstrapper.doneBootstrapping = ->
 		bootstrapper.bootstrapped = true
 		resolve(userConfig)
+		# If we're still using an old api key we can try to exchange it for a valid device key
+		if userConfig.apiKey?
```

Comment:
This will be done after the `resolve()` above... shouldn't we exchange the key before continuing? Also, this doesn't write to config.json after deleting userConfig.apiKey, so the file will not be modified